google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPoolOptions.java

```diff
@@ -20,6 +20,8 @@
 
 /** Options for the session pool used by {@code DatabaseClient}. */
 public class SessionPoolOptions {
+  private static final int DEFAULT_MAX_SESSIONS = 2000;
+  private static final ActionOnExhaustion DEFAULT_ACTION = ActionOnExhaustion.FAIL;
```

Comment:
I'm not convinced by FAIL as the default: it could lead to unpredictable exceptions if the user application briefly blips up in concurrency, rather than a simple latency increase. How about keeping BLOCK, but having a log warning when this happens?